src/envoy/http/authn/filter_context_test.cc

```diff
@@ -0,0 +1,123 @@
+/* Copyright 2018 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "src/envoy/http/authn/filter_context.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "src/envoy/http/authn/context.pb.h"
+#include "src/envoy/http/authn/test_utils.h"
+#include "test/test_common/utility.h"
+
+using testing::StrictMock;
+
+namespace iaapi = istio::authentication::v1alpha1;
+
+namespace Envoy {
+namespace Http {
+namespace IstioAuthN {
+namespace {
+
+class FilterContextTest : public testing::Test {
+ public:
+  virtual ~FilterContextTest() {}
+
+  void SetUp() override {
+    filter_context_.reset(new StrictMock<MockFilterContext>);
+  }
+
+  std::unique_ptr<StrictMock<MockFilterContext>> filter_context_;
```

Comment:
Just drectly use MockFilterContext or NickMock<MockFilterContext>  

Not need to use unique_ptr

Reply:
Done